### PR TITLE
Boolean Flag Proposed In Kong PR Change #3703

### PR DESCRIPTION
**Pull request by jeremyjpj0916**
### Summary

This is in direct relation to my proposed PR here: https://github.com/Kong/kong/pull/3703

New flag hide_groups_header to turn off passing the API providers potentially extraneous growing `X-Consumer-Groups` header list.

### Full changelog

*Edited /app/plugins/acl.md to include new flag field and related information about its use.

### Issues resolved

Syncs with PR https://github.com/Kong/kong/pull/3703

### Checklist:
- [ ] Ready to be merged  -> Not until the related PR is released into Kong functionality!

### CODE REVIEW

**Comment by thibaultcha** [comment]
@jeremyjpj0916 I changed the base branch to `release/ce-0.14.1` here - if you can, please rebase your commit on top of this branch. Thanks!

**Comment by jeremyjpj0916** [comment]
Asked a few people here and none knows the process to achieve this(Last 2 years I have used svn mostly, sad times), I will just open a PR pointing to the proper branch 14.1. Know any good ELI5 blogs for what you have asked twice of me I lack knowledge for?
